client/light/src/call_executor.rs

```diff
@@ -230,62 +211,41 @@ pub fn check_execution_proof<Header, E, H>(
 		H: Hasher,
 		H::Out: Ord + codec::Codec + 'static,
 {
-	check_execution_proof_with_make_header::<Header, E, H, _>(
+	check_execution_proof_with_make_header::<Header, E, H>(
```

Comment:
Doesn't this break compatibility between proof formats? Are there no consequences?